### PR TITLE
[flutter_tools] use a single fake implementation of HttpClient

### PR DESCRIPTION
**Pull request by jonahwilliams**
Replace all Mock/Fake http clients with a single FakeHttpClient. This works similar to the fake process manager, except that it does not enforce a strict order.

example API:

```dart

var client = FakeHttpClient.any(); // Returns a 200 empty request for everything

// Responds to a get request to localhost:8181 with a 201 with 3 bytes, other requests return
// a StateError
var client2 = FakeHttpClient.list([
  FakeRequest(Uri.parse('localhost:8181'), response: FakeResponse(body: [1, 2, 3], statusCode: 201))),
]);
```

https://github.com/flutter/flutter/issues/71511

### CODE REVIEW

**Review by jmagman — COMMENTED** [review]
Is there any way to leverage [MockClient](https://pub.dev/documentation/http/latest/testing/MockClient-class.html) instead of maintaining our own?

**Comment by jonahwilliams** [comment]
> Is there any way to leverage MockClient instead of maintaining our own?


That is a mock of the package:http client, not the dart:io http client.

File: packages/flutter_tools/test/general.shard/testbed_test.dart [hunk 1]
```diff
@@ -70,7 +70,7 @@ void main() {
         final HttpClientRequest request = await client.getUrl(null);
         final HttpClientResponse response = await request.close();
 
-        expect(response.statusCode, HttpStatus.badRequest);
+        expect(response.statusCode, HttpStatus.ok);
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/testbed_test.dart:73]: Correct, I changed the default to OK as it did not seem that important.

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**